Library/Homebrew/dev-cmd/typecheck.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+require "cli/parser"
+
+module Homebrew
+  module_function
+
+  def typecheck_args
+    Homebrew::CLI::Parser.new do
+      usage_banner <<~EOS
+        `typecheck`
+
+        Check for typechecking errors using Sorbet.
+      EOS
+      switch "--quiet",
+             description: "Silence all non-critical errors."
+      flag   "--dir=",
+             description: "Typecheck all files in a specific directory."
+      flag   "--file=",
+             description: "Typecheck a single file."
+      flag   "--ignore=",
+             description: "Ignores input files that contain the given string"\
+                          " in their paths (relative to the input path passed to Sorbet)."
+      conflicts "--dir", "--file"
+
+      max_named 0
+    end
+  end
+
+  def typecheck
+    args = typecheck_args.parse
+
+    Homebrew.install_bundler_gems!
+
+    HOMEBREW_LIBRARY_PATH.cd do
+      ENV["HOMEBREW_SORBET"] = "1"
+      srb_exec = "bundle exec srb tc"
```

Comment:
```suggestion
      srb_exec = ["bundle", "exec", "srb", "tc"]
```

then add to array below and use `system(*srb_exec)` to execute. This will handle e.g. spaces in filenames and avoid spawning a new shell.